components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java

```diff
@@ -55,6 +55,20 @@ public static String getQueryParameter(MessageContext messageContext, String key
         return null;
     }
 
+    /**
+     * Extracts the value set for the patch parameter with the given key.
+     *
+     * @param messageContext message context to extract the parameter from
+     * @param key            the key defined in the uri template
+     * @return the resolved value from the url. Returns null if not present.
+     */
+    public static String getPathParameter(MessageContext messageContext, String key){
+        if (Objects.nonNull(messageContext.getProperty(RESTConstants.REST_URI_VARIABLE_PREFIX + key))) {
+            return messageContext.getProperty(RESTConstants.REST_URI_VARIABLE_PREFIX + key).toString();
```

Comment:
Shall we assign "messageContext.getProperty(RESTConstants.REST_URI_VARIABLE_PREFIX + key)" to a variable and do the processing without retrieving from the msg context two times.

Reply:
done